widgetastic_manageiq/__init__.py

```diff
@@ -2780,62 +2868,49 @@ def get_all(self, surf_pages=False):
         Returns: all entities (QuadIcon/etc.) displayed by view
         """
         if not surf_pages:
-            return [self.parent.entity_class(parent=self, name=name) for name in self.entity_names]
+            return [self.parent.entity_class(parent=self, id=id) for id in self.entity_ids]
         else:
             entities = []
             for _ in self.paginator.pages():
-                entities.extend([self.parent.entity_class(parent=self, name=name)
-                                for name in self.entity_names])
+                entities.extend([self.parent.entity_class(parent=self, id=id)
+                                for id in self.entity_ids])
             return entities
 
-    def get_entities(self, by_partial_name, surf_pages=False):
-        """ obtains all matched entities like QuadIcon displayed by view
-        Args:
-            by_partial_name (str): only entities which match to by_name will be returned
-            surf_pages (bool): current page entities if False, all entities otherwise
-
-        Returns: all matched entities (QuadIcon/etc.) displayed by view
-        """
-        entities = self.get_all(surf_pages)
-        remaining_entities = []
-        for entity in entities:
-            if by_partial_name in entity.name:
-                remaining_entities.append(entity)
-            # todo: by_type and by_regexp will be implemented later if needed
-        if not remaining_entities:
-            raise ItemNotFound("Entities matching to {name} "
-                               "aren't found on this page".format(name=by_partial_name))
-        return remaining_entities
-
-    def get_entity(self, by_name, surf_pages=False):
+    def get_entity(self, surf_pages=False, **keys):
```

Comment:
this is a **breaking** api change, we need to correctly handle it - 3rd party users may be taken by complete surprise

Reply:
I don't think there are any users apart form us :laughing:

Reply:
it won't be difficult to me to add support of api. However, I'm not sure it is necessary.
@psav, what do you think ?

Reply:
I don't think it's necessary right now